### ISSUE TITLE
Inference Questions

### ISSUE THREAD

**Issue by dani-ture** [mathpluscode/CineMA#28]
Hello! Congratulations on your fantastic work, and many thanks for sharing it.

I'm a beginner in the field of medical imaging, and I would like to try the classification models using my **cine short-axis** images. However, I am facing two issues:

- I do not have ground-truth masks, so I cannot crop the images around the center of the left ventricle (LV).
- I do not have information about which frames belong to the end-systole (ES) and end-diastole (ED). 

How would you recommend proceeding? I’ve considered a few possible approaches:

**Cropping**:
- **Option 1**: The heart is already approximately centered in my images, so I could perform a fixed center crop with the recommended dimensions after resampling.
- **Option 2**: Manually locate the LV and crop accordingly (not really feasible).
- **Option 3**: Use Option 1 first, then run your segmentation model and finally crop and pad again based on the predicted LV location.

**Selecting the ES and ED frames**:
- **Option 1**: Visual selection (again not really feasible, and I could easily be off by 1–2 frames)
- **Option 2**: Run your segmentation model and, for each slice, select the frame in which the LV volume is maximal for ED and minimal for ES.

What do you think? I would really appreaciate your advice. Thank you so much!

**Comment by mathpluscode** [comment]
Hi @dani-ture, thanks for your interest in our work! I think your suggestions are already great! I have added my comments in the end.

Otherwise, I would potentially add one more point. All your questions involve in segmentation models. We have released the segmentations models trained on three datasets: ACDC, M&Ms, M&Ms2, for each data we have trained three models with three different seeds. This means that there are actually nine segmentation models available, their performance may vary on your dataset, you could potentially try these models (should be just changing model checkpoints).

I would first determine which dataset (ACDC, M&Ms, M&Ms2) is the closest to your dataset such that the model trained on it delivers the best performance. I would try M&Ms/M&Ms2 first, as the training data set is larger and their performance on unseen Kaggle/Rescan datasets is better:

<img width="409" height="348" alt="Image" src="https://github.com/user-attachments/assets/c57e0cbf-e879-4049-9cf1-0c937472f973" />

Then once you select the training/fine-tuning data of the segmentation models, as there are three different seeds, you can obtain three different segmentation masks to perform ensembles. For example, if you want to determine ED/ES frames, you can average the predicted LV volume across three models.

Hope these responses are helpful!
Happy new year!


> I do not have ground-truth masks, so I cannot crop the images around the center of the left ventricle (LV).

I would suggest a center cropping, which is likely the easiest and most efficient way, which is your option 1. Maybe do visualise the cropped region to see if sensible and in case that certain images are not correctly cropped, consider a manual fix.

Otherwise, as you suggested in option 3, you can use the released segmentation models to segment the images, if visualisation show that the LV segmentations are sensible, you can then derive the LV center and crop it.

> I do not have information about which frames belong to the end-systole (ES) and end-diastole (ED).

I would go for option 2, this is how the demo works on UKB data.
